botocore/signers.py

```diff
@@ -196,6 +208,65 @@ def generate_presigned_url(self, request_dict, expires_in=3600,
         return request.url
 
 
+class CloudFrontSigner(object):
+
+    @staticmethod
```

Comment:
Well, it does not need anything else from other class member, but it still tightly couples with the functionality of this class, so I put it in this class, as a static method. By the way, there is a [similar function in Boto, also organized as a static method](https://github.com/boto/boto/blob/develop/boto/cloudfront/distribution.py#L680-L690).

For the "why not internal" part, I think it doesn't harm to also provide this helper for outsider, for what it's worth. If you prefer to name it as _b64encode() or even __b64encode(), just let me know.